src/svg2pdf.js

```diff
@@ -151,16 +153,103 @@ SOFTWARE.
     return pathSegList;
   };
 
+  function StyleSheets(rootSvg, loadExtSheets) {
+    this.rootSvg = rootSvg;
+    this.loadExtSheets = loadExtSheets;
+    this.sheets = [];
+    this.selectorMap = {};
+  }
+
+  StyleSheets.prototype.getParsedSheets = function () {
+    if (this.sheets.length === 0) {
+      var sheetTexts = [];
+
+      var styleTags = this.rootSvg.getElementsByTagName("style") || [];
+      for (styleTag of styleTags) {
+        sheetTexts.push(styleTag.textContent);
+      }
+
+      if (this.loadExtSheets || true) {
+        var paths = [];
+        var links = this.rootSvg.getElementsByTagName("link");
+        for (var link of links) {
+          if (link.getAttribute("rel") === "stylesheet") {
+            paths.push(link.getAttribute("href"));            
+          }
+        }
+
+        for (var node of this.rootSvg.ownerDocument.childNodes) {
+          if (node.nodeName === "xml-stylesheet") {
+            paths.push(node.data.match(/href=[\"|\'|\`|\`].*?[\"|\'|\`|\`]/)[0].split("=")[1].slice(1,-1));
+          }
+        }
+
+        for (var path of paths) {
+          var response = StyleSheets.loadSheet(path);
+          if (!(response instanceof Error)) {
+            sheetTexts.push(response);
+          }
+        }
+      }
+
+      for (var text of sheetTexts) {
+        var sheet = Css.parse(text, {silent: true});
+        this.sheets.push(sheet);
+      }
+    }
+    return this.sheets;
+  }
+
+  StyleSheets.loadSheet = function (url) {
+    const request = new XMLHttpRequest();
```

Comment:
We could try and use the [styleSheets API](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/styleSheets) for this. Don't know if this works, though.

Reply:
I retried it and couldn't find any styleSheets in the debug.html DOM...